platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -34,35 +33,56 @@
  */
 public class DiagnosticToProblemListener implements DiagnosticListener<JavaFileObject> {
 
-    private final InternalProblemReporter problemReporter;
+    private final ProblemReporter problemReporter;
 
-    public DiagnosticToProblemListener(Problems problemsService) {
-        this.problemReporter = ((InternalProblems) problemsService).getInternalReporter();
+    public DiagnosticToProblemListener(ProblemReporter problemReporter) {
+        this.problemReporter = problemReporter;
     }
 
     @Override
     public void report(Diagnostic<? extends JavaFileObject> diagnostic) {
         String message = diagnostic.getMessage(Locale.getDefault());
         String label = mapKindToLabel(diagnostic.getKind());
-        String resourceName = diagnostic.getSource() != null ? diagnostic.getSource().toUri().toString() : null;
-        Integer line = Math.toIntExact(diagnostic.getLineNumber());
-        Integer column = Math.toIntExact(diagnostic.getColumnNumber());
-        Integer length = Math.toIntExact(diagnostic.getEndPosition() - diagnostic.getStartPosition());
+
+        String resourceName = diagnostic.getSource() != null ? getPath(diagnostic.getSource()) : null;
+        int line = Math.toIntExact(diagnostic.getLineNumber());
+        int column = Math.toIntExact(diagnostic.getColumnNumber());
```

Comment:
❓ The line and column values are 1-indexed, right?

Reply:
Checked (as it's undocumented), and they are